src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -378,13 +322,14 @@ private void onFill(IForgeRegistry<?> registry)
 
     private void addEntries(RegisterEvent event)
     {
-        if (event.getRegistryKey().equals(this.registryKey))
+        if (event.getRegistryKey() == this.registryKey)
         {
             this.seenRegisterEvent = true;
             for (Entry<RegistryObject<T>, Supplier<? extends T>> e : entries.entrySet())
             {
                 event.register(this.registryKey, e.getKey().getId(), () -> e.getValue().get());
-                e.getKey().updateReference(event);
+                // Throw on missing registry here because absence of the registry indicates something is very wrong.
```

Comment:
No, because we're inside a block of `event.getRegistryKey() == this.registryKey`

If we pass that check it means the registry exists (as the RegisterEvent has been fired for it).

Reply:
https://discord.com/channels/313125603924639766/1131130393144336385/1136025733463023718